accounting/gst.js

```diff
@@ -150,11 +189,79 @@ async function generateB2bData(invoices) {
 }
 
 async function generateB2clData(invoices) {
-  return [];
+  const b2cl = [];
+
+  invoices.forEach(async (invoice) => {
+
+    const stateInvoiceRecord = {
+      pos: posMap[invoice.place.toUpperCase()],
+      inv: []
+    };
+
+    const invRecord = {
+      inum: invoice.invNo,
+      idt: DateTime.fromFormat(invoice.invDate, 'yyyy-MM-dd').toFormat(
+        'dd-MM-yyyy'
+      ),
+      val: invoice.invAmt,
+      itms: [],
+    }
+
+    let items = await frappe.db
+      .knex('SalesInvoiceItem')
+      .where('parent', invRecord.inum);
+
+    items.forEach((item) => {
+      const itemRecord = {
+        num: item.hsnCode || 0,
+        itm_det: {
+          txval: item.baseAmount,
+          rt: GST[item.tax],
+          csamt: 0,
+          iamt: ((invoice.rate || 0) * item.baseAmount) / 100,
+        },
+      };
+
+      invRecord.itms.push(itemRecord);
+    });
+
+    const stateRecord = b2cl.find((b) => b.pos === posMap[invoice.place]);
+
+    if (stateRecord) {
+      stateRecord.inv.push(invRecord);
+    } else {
+      stateInvoiceRecord.inv.push(invRecord);
+      b2cl.push(stateInvoiceRecord);
+    }
+
+  })
+
+  return b2cl;
 }
 
 async function generateB2csData(invoices) {
-  return [];
+  const b2cs = [];
+
+  invoices.forEach(async (invoice) => {
+
+    const pos = invoice.place.toUpperCase();
+
+    const invRecord = {
+      "sply_ty": invoice.inState ? "INTRA" : "INTER",
+      "pos": posMap[pos],
+      "typ": "OE",
```

Comment:
Let's add a comment here on what OE type along with some reference link so in case anyone else starts wondering what OE type is they will have their answer here inline.

Reply:
Done 💯